3d_armor/api.lua

```diff
@@ -117,6 +119,14 @@ armor.register_armor = function(self, name, def)
 		end
 		return armor:equip(player, itemstack)
 	end
+	-- The below is a very basic check to try and see if a material name exists as part
+	-- of the item name. However this check is very simple and just checks theres "_something"
+	-- at the end of the item name and logging an error to debug if not.
+	local check_mat_exists = string.match(name, "%:.+_(.+)$")
```

Comment:
Would it still work if `hardened_steel` is used as material?

Reply:
Ill work on the other comments but for this one thats a no, thats the current limitation inside 3d_armor. The string match line above is ripped straight from the current api although I adjusted it a  little for context:
https://github.com/minetest-mods/3d_armor/blob/master/3d_armor/api.lua#L261
```
			local mat = string.match(item, "%:.+_(.+)$")
```

It was what I was tossing up with by going fully to armor groups as then more exotic names will work and much cleaner. In the end I thought moving to armor groups might be better in a more consedered anf formal way ie specific version jump. As supporting both the legacy and new groups turns into a bunch of code mess....

Reply:
Alright I see. It's already a limitation, thus this code is okay. Underscore-separated materials support can be added/fixed when it's needed - in another PR.

Reply:
I might have spoken to soon on this one, I'm testing something out and with the streamlining I did before plus your cool new worn_key line SmallJoker I think I can add group support + legacy + unknown with one extra line.

I haven't fully tested this other than some basic checking using lua demo this appears it should work:
```
           for loc,item in pairs(worn_armor) do             
                  local item_mat_leg = string.match(item, "%:.+_(.+)$")
                  local item_mat_grp = string.match(minetest.serialize(minetest.registered_items[item].groups),"armor_m_(.*)=(.-)")
                  local worn_key = item_mat_grp or item_mat_leg or "unknown"

```
This would mean some updates to the readme but I think for 1 extra line it would start the transition to groups. The way the groups are defined is using the name so

armor_m_hardened_steel=1 would be added to the group definitions for the armor item. The above string match would then pull out only the "hardened_steel". So long as someone didn't name their material armor_m_hardened=steel=1 which I think would cause other errors in the engine the above should always work. 

Priority order would be "group name">>"end of item name">>"unknown"

So long as you're open to a little more checking for yourself and the group name for materials definition looks okay (armor_m_name_of_material), I'm happy to do the above. I unfortunately can't use the existing group names as then you can use wood helmets for example to make more wood helmets or sticks.

I'll get onto those other fixes and updates :) .